react/Header/UserAccountMenu/UserAccountMenu.js

```diff
@@ -78,7 +78,7 @@ export default function UserAccountMenu({ locale, authenticationStatus, linkRend
           linkRenderer({
             'data-analytics': 'header:saved+jobs',
             className: `${styles.item} ${styles.subItem}`,
-            href: '/my-activity/saved-jobs',
+            href: authenticationStatus !== AUTHENTICATED ? '/sign-in?returnUrl=/my-activity/saved-jobs' : '/my-activity/saved-jobs',
```

Comment:
Could this ternary be factored out into a function to get rid of the URL duplication? e.g. `urlForAuthStatus(authenticationStatus, '/my-activity/saved-jobs')` (happy for you to come up with something better, that's just to illustrate my point)